### PR TITLE
Uniswap V3 - Schema 1.2.1

### PR DESCRIPTION
**Pull request by steegecs**

### CODE REVIEW

**Comment by this-username-is-taken** [comment]
Remove the `.DS_Store`s. I think they should be ignored automatically since this is included in `.gitignore`. You probably want to look into why it keeps getting committed and make sure it doesn't happen again in the future.

File: subgraphs/uniswap-v3/src/common/constants.ts [hunk 1]
```diff
@@ -1,99 +1,114 @@
...
-export const WETH_ADDRESS = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
-export const USDC_WETH_PAIR = '0xb4e16d0168e52d35cacd2c6185b44281ec28c9dc' // created 10008355
-export const DAI_WETH_PAIR = '0xa478c2975ab1ea89e8196811f51a7b7ade33eb11' // created block 10042267
-export const USDT_WETH_PAIR = '0x0d4a11d5eeaac28ec3f61d100daf4d40471f1852' // created block 10093341
+export const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
 
 export const DEFAULT_DECIMALS = 18;
 export const USDC_DECIMALS = 6;
-export const USDC_DENOMINATOR = BigDecimal.fromString("1000000");
+export const USDC_DENOMINATOR = BigDecimal.fromString("100000");
```
Comment (this-username-is-taken) [inline subgraphs/uniswap-v3/src/common/constants.ts:77]: This should be `1000000` (10^6)

File: subgraphs/uniswap-v3/src/common/creators.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+  // Get the value in USD of the deposit
+  let amountUSD = amount0Converted.times(token0.lastPriceUSD!).plus(amount1Converted.times(token1.lastPriceUSD!));
+
+  // reset tvl aggregates until new amounts calculated
+  protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.minus(pool.totalValueLockedUSD);
+
+  // Update pool balances adjusted for decimals and not adjusted
+  pool.inputTokenBalances = [pool.inputTokenBalances[0].plus(amount0), pool.inputTokenBalances[1].plus(amount1)];
+  poolAmounts.inputTokenBalances = [poolAmounts.inputTokenBalances[0].plus(amount0Converted), poolAmounts.inputTokenBalances[1].plus(amount1Converted)];
```
Comment (this-username-is-taken) [inline subgraphs/uniswap-v3/src/common/creators.ts:135]: Just a stylistic thing, usually in Typescript/JS this is written as:
```
poolAmounts.inputTokenBalances = [
  poolAmounts.inputTokenBalances[0].plus(amount0Converted), 
  poolAmounts.inputTokenBalances[1].plus(amount1Converted)
];
```
for better readability

File: subgraphs/uniswap-v3/src/common/creators.ts [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+  // Get the total value locked in USD
+  pool.totalValueLockedUSD = poolAmounts.inputTokenBalances[0].times(token0.lastPriceUSD!).plus(poolAmounts.inputTokenBalances[1].times(token1.lastPriceUSD!));
```
Comment (this-username-is-taken) [inline subgraphs/uniswap-v3/src/common/creators.ts:138]: Same here:
```
pool.totalValueLockedUSD = poolAmounts.inputTokenBalances[0]
  .times(token0.lastPriceUSD!)
  .plus(poolAmounts.inputTokenBalances[1].times(token1.lastPriceUSD!));
```

File: subgraphs/uniswap-v3/config/_paramConfig.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+import { toLowerCase } from "../src/common/utils/utils";
+
+export namespace Protocol {
+  export const UNISWAPV3 = "Uniswap V3";
+}
+
+// Choose which protocol you are indexing. The deployed network will already be determined
+let PROTOCOL_NAME_TEMP = Protocol.UNISWAPV3;
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (this-username-is-taken) [inline subgraphs/uniswap-v3/config/_paramConfig.ts:12]: Divider should be less than or equal to 80 characters per row for readability

File: subgraphs/uniswap-v3/config/_paramConfig.ts [hunk 5]
```diff
@@ -0,0 +1,197 @@
+import { dataSource, BigDecimal, log } from "@graphprotocol/graph-ts";
```
Comment (this-username-is-taken) [inline subgraphs/uniswap-v3/config/_paramConfig.ts:1]: no need to prepend file with underscore

**Review by this-username-is-taken — DISMISSED** [review]
Great work! Looks good in general. Just some minor things to address.

I've approved it. Feel free to merge it in yourself after addressing the CR comments

File: subgraphs/uniswap-v3/src/common/creators.ts (hunk 2 above)
Comment (steegecs) [reply subgraphs/uniswap-v3/src/common/creators.ts:135]: Updated

File: subgraphs/uniswap-v3/src/common/constants.ts (hunk 1 above)
Comment (steegecs) [reply subgraphs/uniswap-v3/src/common/constants.ts:77]: Fixed

File: subgraphs/uniswap-v3/config/_paramConfig.ts (hunk 4 above)
Comment (steegecs) [reply subgraphs/uniswap-v3/config/_paramConfig.ts:12]: Split to 2 rows

File: subgraphs/uniswap-v3/config/_paramConfig.ts (hunk 5 above)
Comment (steegecs) [reply subgraphs/uniswap-v3/config/_paramConfig.ts:1]: Removed underscore

**Pull request merged**